src/discovery/Discovery.ts

```diff
@@ -81,14 +89,35 @@ class Discovery {
   protected gestaltGraph: GestaltGraph;
   protected identitiesManager: IdentitiesManager;
   protected nodeManager: NodeManager;
+  protected taskManager: TaskManager;
 
-  protected discoveryDbPath: LevelPath = [this.constructor.name];
-  protected discoveryQueueDbPath: LevelPath = [this.constructor.name, 'queue'];
-  protected discoveryQueueIdGenerator: DiscoveryQueueIdGenerator;
+  // Protected discoveryDbPath: LevelPath = [this.constructor.name];
+  // protected discoveryQueueDbPath: LevelPath = [this.constructor.name, 'queue'];
+  // protected discoveryQueueIdGenerator: DiscoveryQueueIdGenerator;
   protected visitedVertices = new Set<GestaltKey>();
-  protected discoveryProcess: Promise<void>;
-  protected queuePlug = promise<void>();
-  protected queueDrained = promise<void>();
+
+  protected discoverVertexHandler: TaskHandler = async (
+    ctx,
+    _taskInfo,
+    vertex: GestaltKey,
+  ) => {
+    try {
+      await this.processVertex(vertex, ctx);
+    } catch (e) {
+      if (e instanceof tasksErrors.ErrorTaskStop) {
+        // We need to recreate the task for the vertex
+        await this.scheduleDiscoveryForVertex(vertex);
+        return;
+      }
+      // Aborting a duplicate task is not an error
+      if (e === abortSingletonTaskReason) return;
+      // Destroying tasks is not an error
+      if (e === discoveryDestroyedTaskReason) return;
+      throw e;
+    }
+  };
+  public readonly discoverVertexHandlerId =
```

Comment:
Would this not have the same problem as nodes where the task handler ids collide?

Reply:
Yes, but there is only 1 handler in this domain, so the handler never gets overwritten.

Reply:
Yea, but if we have a hack for #453, then the same hack should be applied consistently. Although before you do so, please check on the CI/CD if the name is applied properly by logging out this property.

And if isn't affecting this domain, update the #453 with this discrepancy. It is concerning.

Reply:
I applied the same fix as for `NodeManager`. I'm skipping checking if the `name` is empty on CI. Makes more sense to check that when addressing the issue directly.